### ISSUE TITLE
Search on Eloquent Eager Loading

### ISSUE THREAD

**Issue by timothyrowan** [yajra/laravel-datatables#771]
Population of table works really well. However when I go to filter using the default datatables filter input, I receive a query exemption error:

> QueryException in Connection.php line 729:
> SQLSTATE[42S22]: Column not found: 1054 Unknown column 'laravel_reserved_0.user_id' in 'where clause' (SQL: select count(*) as aggregate from (select '1' as `row_count` from `files` where `attachment_type` = App\Spot and `status` = 7 and (LOWER(`name`) LIKE %%a%% or (select count(1) from `files` as `laravel_reserved_0` where `laravel_reserved_0`.`id` = `laravel_reserved_0`.`attachment_id` and LOWER(`slug`) LIKE %%a%% and `laravel_reserved_0`.`deleted_at` is null) >= 1 or (select count(1) from `users` where `laravel_reserved_0`.`user_id` = `users`.`id` and LOWER(`email`) LIKE %%a%% and `users`.`deleted_at` is null) >= 1 or LOWER(`created_at`) LIKE %%a%% or LOWER(`expires_at`) LIKE %%a%%) and `laravel_reserved_0`.`deleted_at` is null) count_row_table)

``` php
$files = File::spot()->finished()->with('attachment', 'user');
return \Datatables::of($files)->make(true);
```
- Operating System - OSX 10.11
- PHP Version - 7.0.9
- Laravel Version 5.2.45
- Laravel-Datatables Version - 6.19.0

**Comment by timothyrowan** [comment]
forgot to add, 'attachment' is a morph to files

**Comment by underdpt** [comment]
This seems related to my latest post on #698

**Comment by yajra** [comment]
@timothyrowan sorry but I think morph relationship is not yet supported. I think only direct relationship works on the package atm like belongsTo, hasOne, hasMany. I am trying to make a demo on my free time to test this relationships and so far, I only have verified it on 2 relationship demo atm.

https://datatables.yajrabox.com/relation/has-one
https://datatables.yajrabox.com/relation/has-many

**Comment by ethaizone** [comment]
@underdpt You should look this issue. #696 I think this one is same issue as you found. I have this problem too. 

This is not about morph relation but it's about nested relations. For make it's clearly. I mean if we have multi level relation with hasOne, it's error.

**Comment by yajra** [comment]
Should be fixed on v6.22.0. Thanks!
